src/com/lftechnology/java/training/dipak/employeemanagement/dao/EmployeeDao.java

```diff
@@ -0,0 +1,251 @@
+
+package com.lftechnology.java.training.dipak.employeemanagement.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.lftechnology.java.training.dipak.employeemanagement.api.EmployeeApi;
+import com.lftechnology.java.training.dipak.employeemanagement.domain.Employee;
+
+import com.mysql.jdbc.exceptions.MySQLIntegrityConstraintViolationException;
+
+/**
+ * <p>This class implements the EmployeeApi and has definition to all the
+ * overwritten methods.</p>
+ * 
+ * @author Dipak Thapa<dipakthapa@lftechnology.com>
+ */
+public class EmployeeDao implements EmployeeApi {
+
+	private static final Logger LOGGER =
+		Logger.getLogger(EmployeeDao.class.getName());
+
+	/**
+	 * <p>This method is used to add a row in database. The data is obtained
+	 * from the Employee object. This method returns the number of rows affected
+	 * in the database</p>
+	 * 
+	 * @param e
+	 * @return count
+	 */
+	@Override
+	public int addEmployee(Employee e) {
+
+		Connection con = DbConnection.setConnection();
+		int count = 0;
+		if (con == null) {
+			LOGGER.info("Connection Setup Failed");
+			return count;
+		}
+		PreparedStatement pst = null;
+		String sql =
+			"insert into employee (username,password,isTerminated,fullname,department,address,role) values(?,?,?,?,?,?,?)";
+		try {
+			pst = con.prepareStatement(sql);
+
+			pst.setString(1, e.getUserName());
+			pst.setString(2, e.getPassword());
+			pst.setBoolean(3, e.getIsTerminated());
+			pst.setString(4, e.getFullName());
+			pst.setString(5, e.getDepartment());
+			pst.setString(6, e.getAddress());
+			pst.setString(7, e.getRole().toString());
+
+			System.out.println(pst);
+
+			count = pst.executeUpdate();
+
+		}
+		catch (MySQLIntegrityConstraintViolationException e1) {
+			LOGGER.severe(
+				"User name already exits. Please enter another username.");
+		}
+		catch (SQLException sq) {
+			LOGGER.log(Level.SEVERE, "Exception::{0}", sq);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Exception::{0}" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)